src/Api/HL/Controller/ITILController.php

```diff
@@ -390,37 +503,93 @@ public function deleteItem(Request $request): Response
+        } else {
+            $schema = $this->getKnownSchema($subitem_type);
+        }
+        return $schema;
+    }
+
     /**
-     * @param CommonITILObject $item
-     * @return array{type: string, item: array<string, mixed>}[]
+     * Get the timeline items for a given item
+     * @param CommonITILObject $item The item to get the timeline items for
+     * @param Request $request The original request
+     * @param array $subitem_types The subitem types to include or all if empty
+     * @return array
      */
-    private function getCleanTimelineItems(CommonITILObject $item): array
-    {
-        $timeline = $item->getTimelineItems();
-        $items = [];
-        // Only keep certain properties
-        foreach ($timeline as &$timeline_item) {
-            $t_item = [
-                '_itemtype' => $this->getSubitemFriendlyType($item, $timeline_item['type']),
-            ];
-            $allowed_props = [
-                'id', 'content', 'uuid', 'date', 'users_id', 'users_id_editor', 'is_private',
-                'actiontime', 'begin', 'end', 'state', 'users_id_tech', 'groups_id_tech',
-                'date_mod', 'date_creation', 'filepath'
-            ];
-            foreach ($allowed_props as $prop) {
-                if (array_key_exists($prop, $timeline_item['item'])) {
-                    $t_item[$prop] = $timeline_item['item'][$prop];
-                }
+    private function getITILTimelineItems(CommonITILObject $item, Request $request, array $subitem_types = []): array
+    {
+        $subitem_types = empty($subitem_types) ? ['Followup', 'Task', 'Document', 'Solution', 'Validation'] : $subitem_types;
+        $results = [];
+        foreach ($subitem_types as $subitem_type) {
+            $filters = $this->getTimelineItemFilters($item, $request, $subitem_type);
+            $schema = $this->getKnownSubitemSchema($item, $subitem_type);
+            if ($schema === null) {
+                continue;
             }
-            if (isset($t_item['filepath'])) {
-                // Replace internal path with external path
-                $front_path = 'front/document.send.php?docid=' . $timeline_item['item']['id'] . '&' . $item::getForeignKeyField() . '=' . $item->getID();
-                $t_item['filepath'] = Html::getPrefixedUrl($front_path);
+            $more_results = true;
+            $max_checks = 10;
+            $check = 1;
+
+            /** @var class-string<CommonDBTM> $schema_itemtype */
+            $schema_itemtype = $schema['x-itemtype'];
+            if (!$schema_itemtype::canView()) {
+                continue;
+            }
+            if (array_key_exists('is_private', $schema['properties']) && !\Session::haveRight($schema_itemtype::$rightname, $schema_itemtype::SEEPRIVATE)) {
+                $filters .= ';is_private==0';
+            }
+
+            while ($check < $max_checks && $more_results) {
+                $subitem_results = Search::searchBySchema($schema, [
```

Comment:
There is a `start` parameter missing here. Each iteration will search for same items.

Also, I wonder if we should limit result. IMHO, it is unlikely to have an ITIL item that have more than 1000 subitems of the same type, and in such case, we cannot even be sure that it will not fail before reaching this limit.